src/main/java/org/junit/rules/Timeout.java

```diff
@@ -152,7 +151,7 @@ protected Statement createFailOnTimeoutStatement(
 
     public Statement apply(Statement base, Description description) {
         try {
-            return createFailOnTimeoutStatement(base);
+            return createFailOnTimeoutStatement(base, description);
```

Comment:
Couldn't you just add a `Description` field and set it here, and then pick up the value of that field in `createFailOnTimeoutStatement()`? We wouldn't expect a `Timeout` instance to be shared between two test methods, so that should be safe. Ugly, but safe :-)